tests/bwc/test_rolling_upgrade.py

```diff
@@ -309,3 +150,178 @@ def _test_rolling_upgrade(self, path, nodes):
+        new_shards = 0
+
+        c.execute("select name from sys.users order by 1")
+        self.assertEqual(c.fetchall(), [["arthur"], ["crate"]])
+
+        c.execute("select * from sys.privileges order by ident")
+        self.assertEqual(
+            c.fetchall(),
+            [['TABLE', 'arthur', 'crate', 'doc.t1', 'DENY', 'DQL'],
+                ['CLUSTER', 'arthur', 'crate', None, 'GRANT', 'DQL']])
+
+        c.execute("REFRESH TABLE doc.t1")
+        c.execute('''
+            SELECT type, AVG(value)
+            FROM doc.t1
+            GROUP BY type
+        ''')
+        c.fetchall()
+        # Ensure aggregation with different intermediate input works, this was an regression for 4.1 <-> 4.2
+        c.execute('''
+            SELECT type, count(distinct value)
+            FROM doc.t1
+            GROUP BY type
+        ''')
+        c.fetchall()
+
+        # Ensure scalar symbols are working across versions
+        c.execute('''
+            SELECT type, value + 1
+            FROM doc.t1
+            WHERE value > 1
+            LIMIT 1
+        ''')
+        c.fetchone()
+        c.execute('SELECT type, value + 1 FROM doc.v1 WHERE value > 1 LIMIT 1')
+        c.fetchone()
+
+        # Ensure match queries work. Table level dedicated index column mapping has been changed in 5.4.
+        c.execute('''
+            SELECT value, title, author
+            FROM doc.t1
+            WHERE MATCH(composite_nested_ft, 'matchMe')
+            ORDER BY value
+        ''')
+        res = c.fetchall()
+        self.assertEqual(res, [
+            [1.0, "matchMe title", {"name": "no match name"}],
+            [2.0, "no match title", {"name": "matchMe name"}],
+        ])
+
+        # Dynamically added empty arrays and ignored object sub-columns are indexed with special prefix starting from 5.5
+        # Ensure that reading such columns work across all versions.
+        # Related to https://github.com/crate/crate/commit/278d45f176e7d1d3215118255cd69afd2d3786ee
+        c.execute('''
+            SELECT author, o['dyn_ignored_subcol']
+            FROM doc.t1
+            WHERE title = 'prefix_check'
+        ''')
+        res = c.fetchall()
+        self.assertEqual(len(res), 1)
+        self.assertEqual(res[0][0], {'dyn_empty_array': []})
+        self.assertEqual(res[0][1], 'hello')
+
+        # Ensure that inserts are working while upgrading
+        c.execute(
+            "INSERT INTO doc.t1 (type, value, title, author) VALUES (3, 3, 'some title', {name='nothing to see, move on'})")
+
+        # Ensure that inserts, which will create a new partition, are working while upgrading
+        c.execute("INSERT INTO doc.parted (id, value) VALUES (?, ?)", [idx + 10, idx + 10])
+        new_shards += shards_per_partition
+
+        # Ensure table/partition versions created are correct
+        if new_node.version >= (5, 0, 0):
+            c.execute("insert into doc.t2 (a, b) values (?, ?)", [idx, idx])
```

Comment:
Is `crate.client.exceptions.ProgrammingError: RelationUnknown[Relation 'doc.t2' unknown]` related?

Before we created table **from** 5

```
if int(path.from_version.split('.')[0]) >= 5:
                c.execute('''
                    create table doc.t2 (
```

`new_node.version` reads to me like we are creating it when upgrading **to** 5?